server/base/src/main/java/org/apache/accumulo/server/zookeeper/TransactionWatcher.java

```diff
@@ -104,13 +104,15 @@ public static void cleanup(ServerContext context, String type, long tid)
       final ZooReader reader = context.getZooReaderWriter();
       final Set<Long> result = new HashSet<>();
       final String parent = context.getZooKeeperRoot() + "/" + type;
-      reader.sync(parent);
-      List<String> children = reader.getChildren(parent);
-      for (String child : children) {
-        if (child.endsWith("-running")) {
-          continue;
+      if (reader.exists(parent)) {
```

Comment:
Is it okay to return an empty set if the parent doesn't exist? I'm not sure what the undocumented return value of this method is supposed to represent, so it's not easy to tell if returning an empty set is okay in this case.

Reply:
This is the only spot the method is called:
https://github.com/apache/accumulo/blob/3fd5cad92f9b63ac19e4466f3f2d5237b905262c/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/BulkImportCacheCleaner.java#L50

Looking at that method, I think it is OK.

Reply:
I was looking at that too - I was thinking it might be okay - the comment in the exception handler says that it will just retry, so it could be the same thing as processing nothing with an empty list. There does not seem to be any difference - it looks like it builds a new list each time run is called and nothing is signalling completion.

Also in that exception handling I noticed (well, the ide did) in BulkImportCacheCleaner:[59]
`log.debug("Error reading bulk import live transactions {}", e);` has mismatched arguments - either {} should be removed or something added.

Another thing is that exception handling is swallowing the interrupt exception.

If you don't want to handle those as part of this PR i'll create an issue.

Reply:
I see that too... already done.  I will push up changes.

Reply:
> Another thing is that exception handling is swallowing the interrupt exception.

What are you thinking? Separate catch for the `InterruptedException`, wrap with `RuntimeException` and rethrow?

Reply:
Because this is a runnable - we may not want (or be able to throw an exception) - so at a minimum we should be able to reset the interrupt status with something like:

Also, not sure what info is valuable in the debug statement, listing the tids might make sense - unless its a big list... Without understanding the context I cannot say at this point....

`   } catch (KeeperException e) {
      // we'll just clean it up again later
      log.debug("Error reading bulk import live transactions {}", tids, e);
    } catch(InterruptedException e) {
      // propagate the interrupt status.
      Thread.currentThread().interrupt();
      log.debug("Interrupted while reading bulk import live transactions {}", tids, e);
    }
`
This comes from [Dealing with InterruptedException](https://www.ibm.com/developerworks/library/j-jtp05236/index.html)

Reply:
Updated in 6511c46